x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelActionRequestTests.java

```diff
@@ -68,6 +87,27 @@ protected NamedWriteableRegistry getNamedWriteableRegistry() {
 
     @Override
     protected Request mutateInstanceForVersion(Request instance, Version version) {
-        return instance;
+        InferenceConfigUpdate adjustedUpdate;
+        InferenceConfigUpdate currentUpdate = instance.getUpdate();
+        if (currentUpdate instanceof NlpConfigUpdate nlpConfigUpdate) {
```

Comment:
Could you use a fancy instanceof switch statement here?

```
switch (currentUpdate) {
        case NlpConfigUpdate i -> NlpConfigUpdateTests.mutateForVersion(update, version);
        ....
};
```

Reply:
oh snap, you are correct!